gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -192,7 +193,7 @@ public String toString() {
 
   private volatile boolean isActive = false;
 
-  public DagManager(Config config, boolean instrumentationEnabled) {
+  public DagManager(Config config, JobStatusRetriever jobStatusRetriever, boolean instrumentationEnabled) {
```

Comment:
Is backward compatibility a concern here?

Reply:
It's not a part of the user-facing API, just an internal class. It should be instantiated by dependency injection framework, so even if new dependencies are added, they should be injected automatically.

So, I don't think backward compatibility is important here.